s3/s3_transport/s3_transport_test.cpp

```diff
@@ -0,0 +1,601 @@
+                 const bool debug_flag,
+                 const char *hostname,
+                 const char *bucket_name,
+                 const char *access_key,
+                 const char *secret_access_key,
+                 const char *filename)
+{
+
+
+    printf("%s:%d (%s) [upload thread=%u, seq=%u] writing from file into s3\n",
+            __FILE__, __LINE__, __FUNCTION__, thread_number, thread_number);
+
+    // thread in irods only deal with sequential bytes.  figure out what bytes this thread deals with
+    uint64_t start = thread_number * (file_size / thread_count);
+    uint64_t end = 0;
+    if (thread_number == thread_count - 1) {
+        end = file_size;
+    } else {
+        end = start + file_size / thread_count;
+    }
+
+    std::ifstream ifs;
+
+    ifs.open(filename, std::ios::in | std::ios::binary | std::ios::ate);
+    if (!ifs.good()) {
+        fprintf(stderr, "failed to open file %s\n", filename);
+        return;
+    }
+
+    ifs.seekg(start, std::ios::beg);
+
+    uint64_t current_buffer_size = end - start;
+
+    printf("%s:%d (%s) [[seq=%u]]  [start=%ld][end=%ld][size=%ld]\n",
+            __FILE__, __LINE__, __FUNCTION__, thread_number, start, end, current_buffer_size);
+
+    char *current_buffer;
+    try {
+        current_buffer = new char[current_buffer_size];
+    } catch(std::bad_alloc&) {
+        fprintf(stderr, "failed to allocate memory for buffer.  exiting...\n");
+        return;
+    }
+
+    ifs.read((char*)(current_buffer), current_buffer_size);
+
+    printf("%s:%d (%s) [thread=%u, seq=%u] done reading file\n",
+            __FILE__, __LINE__, __FUNCTION__, thread_number, thread_number);
+
+    /*****************************************
+     * This part actually goes in S3 plugin. *
+     *****************************************/
+
+    s3_transport_config s3_config;
+    s3_config.object_size = file_size;
+    s3_config.number_of_transfer_threads = 20;
+    s3_config.part_size = current_buffer_size;
+    s3_config.bucket_name = bucket_name;
+    s3_config.access_key = access_key;
+    s3_config.secret_access_key = secret_access_key;
+    s3_config.thread_identifier = thread_number;
+    s3_config.debug_flag = debug_flag;
+    s3_config.multipart_flag = use_multipart_flag;
+    s3_config.shared_memory_timeout_in_seconds = 60;
+
+    s3_transport tp1{s3_config};
+    odstream ds1{tp1, s3_prefix + filename};
+
+    if (!ds1.is_open()) {
+        printf("[%d] Open failed.  Exiting...\n", thread_number);
+        return;
+    }
+
+    ds1.seekp(start);
+
+    // doing multiple writes of 1MiB here just to test that that works
+    /*const uint64_t max_write_size = 512*1024*1024;
+    uint64_t write_offset = 0;
+    while (write_offset < current_buffer_size) {
+        uint64_t write_size = std::min(max_write_size, current_buffer_size - write_offset);
+        ds1.write(current_buffer + write_offset, write_size);
+        write_offset += write_size;
+    }*/
```

Comment:
dead?

Reply:
This program is no longer being used and has been removed.